File: tests/claude-plugin-integration/test-repos/go-sample/payment.go
package main

import "fmt"

type PaymentService struct{}

func NewPaymentService() *PaymentService {
	return &PaymentService{}
}

func (ps *PaymentService) ProcessPayment(user *User, amount float64) bool {
	fmt.Printf("Processing $%.2f for %s\n", amount, user.Name)
	return true
}

func (ps *PaymentService) RefundPayment(user *User, amount float64) bool {
	fmt.Printf("Refunding $%.2f to %s\n", amount, user.Name)
	return true
}

File: tests/claude-plugin-integration/test-repos/rust-sample/src/payment.rs
use crate::user::User;

pub struct PaymentService;

impl PaymentService {
    pub fn new() -> Self {
        PaymentService
    }

    pub fn process_payment(&self, user: &User, amount: f64) -> bool {
        println!("Processing ${} for {}", amount, user.name);
        true
    }

    pub fn refund_payment(&self, user: &User, amount: f64) -> bool {
        println!("Refunding ${} to {}", amount, user.name);
        true
    }
}
